File: 덱/JunYoungKr/5430/5430.py
# 함수 R은 배열에 있는 수의 순서를 뒤집는 함수
# D는 첫 번째 수를 버리는 함수. 배열이 비어있는 경우에 호출하면 에러 발생
from collections import deque


# 테스트 케이스 개수 T
T = int(input())

for i in range(T):
    # 수행할 함수 p
    p = input()
    
    # 배열에 들어있는 수의 개수 n
    n = int(input())
    
    queue = deque(input().strip('[]').split(','))
    # print(queue)
    
    # RR일때는 뒤집어 줄 필요가 없기에 홀수일 때만 뒤집기
    flag = 0
    
    # 찾아본 결과 deque는 ['']의 길이를 1로 취급하기 때문에 초기화 해야한다고 함
    if n == 0:
        queue = []
        
    # 수행할 함수의 길이만큼 반복문
    for i in p:
        if i == 'R':
            flag += 1
        elif i == 'D':
            if len(queue) == 0:
                print('error')
                break
            else:
                if flag % 2 == 1:
                    queue.pop()
                else:
                    queue.popleft()
    else:
        if flag % 2 == 1:
            queue.reverse()
        print('[' + ','.join(queue) + ']')
    

File: 덱/DonggyunKim00/5430.js
const fs = require('fs');

const filePath = process.platform === 'linux' ? '/dev/stdin' : 'input.txt';
const input = fs.readFileSync(filePath, 'utf8').toString().trim().split('\n');

class Deque {
  constructor(p, n, arr) {
    this.list = {};
    this.method = p;
    this.head = 0;
    this.tail = n - 1;
    this.isReverse = false;
    this.#init(arr);
  }

  #init(arr) {
    for (let i = 0; i <= this.tail; i++) {
      this.list[i] = arr[i];
    }
  }

  #delete() {
    if (this.isReverse) {
      delete this.list[this.tail];
      this.tail -= 1;
    } else {
      delete this.list[this.head];
      this.head += 1;
    }
  }

  #reverse() {
    this.isReverse = !this.isReverse;
  }

  #size() {
    return this.tail - this.head + 1;
  }

  run() {
    const result = [];
    for (let i = 0; i < this.method.length; i++) {
      const command = this.method[i];

      if (!this.#size() && command === 'D') {
        result.push('error');
        break;
      }

      switch (command) {
        case 'R':
          this.#reverse();
          break;
        case 'D':
          this.#delete();
          break;
      }

      if (i === this.method.length - 1) {
        const values = Object.values(this.list);

        this.isReverse
          ? result.push(JSON.stringify(values.reverse()))
          : result.push(JSON.stringify(values));
      }
    }

    return result[0];
  }
}

const solution = (T, testCase) => {
  const answer = [];
  testCase.forEach(([p, n, arr]) => {
    const P = p.split('');
    const N = Number(n);
    const ARR = JSON.parse(arr);

    const deque = new Deque(P, N, ARR);
    answer.push(deque.run());
  });
  return answer;
};

const T = parseInt(input.shift());
const testCase = input.reduce((acc, cur, idx) => {
  if (idx % 3 === 0) acc.push([]);
  acc[acc.length - 1].push(cur);
  return acc;
}, []);

const answer = solution(T, testCase).join('\n');
console.log(answer);
